refinery/core/tests.py

```diff
@@ -2297,3 +2305,151 @@ def test_create_workflow_engine_bad_group_name(self):
                 str(galaxy_instance.id),
                 "non-existent group name"
             )
+
+
+class InitialSiteStatisticsCreationTest(TestMigrations):
+    migrate_from = '0019_sitestatistics'
+    migrate_to = '0020_create_initial_site_statistics'
+
+    def setUpBeforeMigration(self, apps):
+        public_group = ExtendedGroup.objects.public_group()
+        self.user_a = User.objects.create_user("user_a", "", "user_a")
+        self.user_b = User.objects.create_user("user_b", "", "user_b")
+        self.client.login(username="user_a", password="user_a")
+        self.client.login(username="user_a", password="user_a")
+        self.dataset_a = create_dataset_with_necessary_models(user=self.user_a)
+        self.dataset_b = create_dataset_with_necessary_models()
+        self.dataset_b.share(public_group)
+
+    def test_initial_site_statistics_created_properly(self):
+        initial_site_statistics = SiteStatistics.objects.last()
+
+        self.assertEqual(initial_site_statistics.datasets_uploaded, 2)
+        self.assertEqual(initial_site_statistics.datasets_shared, 1)
+        self.assertEqual(initial_site_statistics.users_created, 3)
+        self.assertEqual(initial_site_statistics.groups_created, 1)
+        self.assertEqual(initial_site_statistics.unique_user_logins, 1)
+        self.assertEqual(initial_site_statistics.total_user_logins, 2)
+        self.assertEqual(initial_site_statistics.total_workflow_launches, 0)
+        self.assertEqual(
+            initial_site_statistics.total_visualization_launches, 0)
+
+
+class SiteStatisticsTests(TestCase):
```

Comment:
This looks suspiciously like a suite of integration tests. I would simplify this by unit testing individual collection methods then write just one of two integration tests.